src/components/KMultiselect/KMultiselect.vue

```diff
@@ -815,9 +817,15 @@ const handleItemSelect = (item: MultiselectItem, isNew?: boolean) => {
   emit('update:modelValue', selectedVals)
 }
 
+const onInputEnter = (): void => {
+  if (!filteredItems.value.length && props.enableItemCreation) {
+    handleAddItem()
+  }
+}
+
 // add an item with `enter`
 const handleAddItem = (): void => {
-  if (!props.enableItemCreation || !filterString.value || !uniqueFilterStr.value) {
+  if (!props.enableItemCreation || !filterString.value || !uniqueFilterStr.value || !props.itemCreationValidator(filterString.value)) {
```

Comment:
non blocking: It seems that we are missing `itemCreationValidator` in `<KSelect>`, shall we align this?

Reply:
Already have a PR for this: https://github.com/Kong/kongponents/pull/2485